packages/flutter/lib/src/foundation/diagnostics.dart

```diff
@@ -1561,9 +1562,76 @@ abstract class DiagnosticsNode {
     if (allowNameWrap)
       data['allowNameWrap'] = allowNameWrap;
 
+    if (delegate.additionalNodeProperties != null) {
+      data.addAll(delegate.additionalNodeProperties(this, delegate));
+    }
+
+    if (delegate.includeProperties) {
+      data['properties'] = propertiesToJsonList(delegate);
+    }
+
+    if (delegate.subtreeDepth > 0) {
+      data['children'] = childrenToJsonList(delegate);
+    }
+
     return data;
   }
 
+  /// Serialize the properties of this node to a JSON list according to the
+  /// configuration provided in the [DiagnosticsSerialisationDelegate].
+  ///
+  /// By default, this will just call [toJsonList] on the return value of
+  /// [getProperties]. Subclasses should override this if their properties
+  /// need special processing.
+  @protected
+  List<Map<String, Object>> propertiesToJsonList(DiagnosticsSerialisationDelegate delegate) {
+    List<DiagnosticsNode> properties = getProperties();
+    properties = delegate.filterProperties != null ? delegate.filterProperties(properties, this, delegate) : properties;
+    return toJsonList(properties, this, delegate);
+  }
+
+  /// Serialize the children of this node to a JSON list according to the
+  /// configuration provided in the [DiagnosticsSerialisationDelegate].
+  ///
+  /// By default, this will just call [toJsonList] on the return value of
+  /// [getChildren]. Subclasses should override this if their children
+  /// need special processing.
+  @protected
+  List<Map<String, Object>> childrenToJsonList(DiagnosticsSerialisationDelegate delegate) {
+    List<DiagnosticsNode> children = getChildren();
+    children = delegate.filterChildren != null ? delegate.filterChildren(children, this, delegate) : children;
```

Comment:
nit: instead of overriding the value of children why not put this expression directly in the the call to toJsonList?